### PR TITLE
Perform one more dynamic partition pruning check

### PR DESCRIPTION
**Pull request by sopel39**
Perform one more dynamic partition pruning check
before split is returned to the engine

### CODE REVIEW

**Comment by findepi** [comment]
cc @losipiuk @alexjo2144

**Review by findepi — APPROVED** [review]
lgtm, but please wait for @raunaqmorarka approval

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java
Comment (raunaqmorarka) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java:357]: There is a check like that at https://github.com/trinodb/trino/pull/9866/files#diff-0434feb95d650801292ef684f874cea122b5274de0a7f71a0c593da59e701d45R166 too
Dynamic filtering of splits was originally moved out of HiveSplitSource and into createInternalHiveSplit on the basis of https://github.com/trinodb/trino/pull/1072#discussion_r353303931
We can probably remove the extra check from createInternalHiveSplit now as there isn't anything too expensive happening between that and this

Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java:357]: btw how expensive is the check by itself?

Comment (raunaqmorarka) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java:357]: There is a small overhead to it but so far i haven't seen it show up on JFR profiles from coordinator

**Comment by sopel39** [comment]
@raunaqmorarka approved in offline conversation

**Pull request merged**
